test/ui/SimpleFilterTest.ts

```diff
@@ -0,0 +1,181 @@
+      test.cmp.toggleCheckbox('foo');
+      expect(test.env.queryController.executeQuery).toHaveBeenCalled();
+    });
+
+    it('should set the field in the query', () => {
+      let simulation = Simulate.query(test3.env);
+      expect(simulation.queryBuilder.groupByRequests).toEqual(jasmine.arrayContaining([
+        jasmine.objectContaining({
+          field: '@field'
+        })
+      ]));
+    });
+
+    it('should set the selected values in the query', () => {
+      test.cmp.selectCheckbox('foo');
+      let simulation = Simulate.query(test.env);
+      expect(simulation.queryBuilder.advancedExpression.build()).toEqual('@field==foo');
+    });
+
+    it('should set the right title depending on values selected', () => {
+      test.cmp.selectCheckbox('foo');
+      expect($$($$(test.cmp.root).find('.coveo-simplefilter-selecttext')).text()).toEqual('foo');
+      test.cmp.selectCheckbox('bar');
+      expect($$($$(test.cmp.root).find('.coveo-simplefilter-selecttext')).text()).toEqual(('FooTitleBar'));
+    });
+
+    it('should handle the backdrop correctly', () => {
+      expect($$(test.cmp.root).find('.coveo-dropdown-background')).not.toBe(undefined);
+      test.cmp.openContainer();
+      expect($$($$(test.cmp.root).find('.coveo-dropdown-background')).hasClass('coveo-dropdown-background-active')).toBe(true);
+      test.cmp.closeContainer();
+      expect($$($$(test.cmp.root).find('.coveo-dropdown-background')).hasClass('coveo-dropdown-background-active')).toBe(false);
+    });
+
+    it('should put all simple filters in the same wrapper if there is more than one', () => {
+      expect(test.cmp.element.parentElement).toEqual(test2.cmp.element.parentElement);
+    });
+
+    it('should not create more than one wrapper for the simple filters', () => {
+      expect($$(test.cmp.root).findAll('.coveo-simplefilter-header-wrapper').length).toEqual(1);
+    });
+
+    it('should link every simple filter to the same backdrop', () => {
+      expect($$(test.cmp.root).findAll('.coveo-dropdown-background').length).toEqual(1);
+    });
+
+    it('should not create more than one backdrop for the simple filters', () => {
+      expect($$(test.cmp.root).findAll('.coveo-dropdown-background').length).toEqual(1);
+    });
+
+    it('should populate the breadcrumb when values are selected', () => {
+      test.cmp.selectCheckbox('bar');
+
+      let args: IPopulateBreadcrumbEventArgs = {
+        breadcrumbs: []
+      };
+      $$(test.env.root).trigger(BreadcrumbEvents.populateBreadcrumb, args);
+      expect(args.breadcrumbs.length).toBe(1);
+    });
+
+    it('should reset checkboxes when clearBreadcrumb is triggered', () => {
+      test.cmp.selectCheckbox('foo');
+      test.cmp.selectCheckbox('bar');
+      let args: IPopulateBreadcrumbEventArgs = {
+        breadcrumbs: []
+      };
+      $$(test.env.root).trigger(BreadcrumbEvents.populateBreadcrumb, args);
+      expect(test.cmp.getSelectedCaptions().length).toBe(2);
```

Comment:
This expect is not necessary since you already test this behaviour in the preceding test. Every test should be able to fail for a single reason :stuck_out_tongue: